### ISSUE TITLE
`RuntimeError` when running tests locally

### ISSUE THREAD

**Issue by ckcollab** [django/daphne#179]
I followed the README and did:

```bash
# inside a fresh Python 3.6 virtualenv
$ pip install -e .[tests]
$ pytest
```

It hangs right here:

```
$ pytest
=================================================== test session starts ====================================================
platform darwin -- Python 3.6.1, pytest-3.4.1, py-1.5.2, pluggy-0.6.0
rootdir: /Users/eric/src/daphne, inifile: setup.cfg
plugins: asyncio-0.8.0, hypothesis-3.45.3
collected 46 items

tests/test_cli.py .........                                                                                          [ 19%]
tests/test_http_request.py
```

Then slowly the failing tests roll in

```
self = <daphne.testing.DaphneTestingInstance object at 0x10834d0b8>

    def __enter__(self):
        # Clear result storage
        TestApplication.delete_setup()
        TestApplication.delete_result()
        # Option Daphne features
        kwargs = {}
        # Optionally enable X-Forwarded-For support.
        if self.xff:
            kwargs["proxy_forwarded_address_header"] = "X-Forwarded-For"
            kwargs["proxy_forwarded_port_header"] = "X-Forwarded-Port"
        if self.http_timeout:
            kwargs["http_timeout"] = self.http_timeout
        # Start up process
        self.process = DaphneProcess(
            host=self.host,
            application=TestApplication,
            kwargs=kwargs,
            setup=self.process_setup,
            teardown=self.process_teardown,
        )
        self.process.start()
        # Wait for the port
        if self.process.ready.wait(self.startup_timeout):
            self.port = self.process.port.value
            return self
        else:
            if self.process.errors.empty():
>               raise RuntimeError("Daphne did not start up, no error caught")
E               RuntimeError: Daphne did not start up, no error caught

daphne/testing.py:57: RuntimeError
========================================== 25 failed, 21 passed in 100.29 seconds ==========================================
```

I *think* I am doing most everything right. I looked at how travis was running tests so I went further and did:

```
$ pip install twisted isort unify flake8 -e .[tests]
```

Same result, though. Tests are passing on Travis so I must be doing something wrong... again hopefully this issue just helps the next person figure out the simple thing they are missing :)

**Comment by andrewgodwin** [comment]
It's probably something about your environment - pass `-s` to pytest to make it not suppress output, but I know the test pass on Travis and on my machine, so I'm not sure what else I can do! The Daphne tests rely on launching processes and stuff, so they are maybe different on OSX?

**Comment by simonw** [comment]
I'm getting this on OS X too:
```
 $ pytest -x -s
============================================= test session starts ==============================================
platform darwin -- Python 3.7.2, pytest-3.10.1, py-1.8.0, pluggy-0.12.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/simonw/Dropbox/Development/daphne/.hypothesis/examples')
rootdir: /Users/simonw/Dropbox/Development/daphne, inifile: setup.cfg
plugins: asyncio-0.10.0, hypothesis-3.88.3
collected 52 items                                                                                             

tests/test_cli.py ............
tests/test_http_request.py F

=================================================== FAILURES ===================================================
______________________________________ TestHTTPRequest.test_bad_requests _______________________________________

self = <test_http_request.TestHTTPRequest testMethod=test_bad_requests>

    def test_bad_requests(self):
        """
        Tests that requests with invalid (non-ASCII) characters fail.
        """
        # Bad path
        response = self.run_daphne_raw(
>           b"GET /\xc3\xa4\xc3\xb6\xc3\xbc HTTP/1.0\r\n\r\n"
        )

tests/test_http_request.py:276: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/http_base.py:66: in run_daphne_raw
    with DaphneTestingInstance() as test_app:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <daphne.testing.DaphneTestingInstance object at 0x104e43208>

    def __enter__(self):
        # Clear result storage
        TestApplication.delete_setup()
        TestApplication.delete_result()
        # Option Daphne features
        kwargs = {}
        # Optionally enable X-Forwarded-For support.
        if self.xff:
            kwargs["proxy_forwarded_address_header"] = "X-Forwarded-For"
            kwargs["proxy_forwarded_port_header"] = "X-Forwarded-Port"
            kwargs["proxy_forwarded_proto_header"] = "X-Forwarded-Proto"
        if self.http_timeout:
            kwargs["http_timeout"] = self.http_timeout
        # Start up process
        self.process = DaphneProcess(
            host=self.host,
            application=TestApplication,
            kwargs=kwargs,
            setup=self.process_setup,
            teardown=self.process_teardown,
        )
        self.process.start()
        # Wait for the port
        if self.process.ready.wait(self.startup_timeout):
            self.port = self.process.port.value
            return self
        else:
            if self.process.errors.empty():
>               raise RuntimeError("Daphne did not start up, no error caught")
E               RuntimeError: Daphne did not start up, no error caught

daphne/testing.py:58: RuntimeError
```
I'll have a poke around and see if I can figure out what's wrong. I might have to run them in a Docker container instead though.

**Comment by simonw** [comment]
My workaround was to run tests in Travis against my own fork: https://travis-ci.org/simonw/daphne/builds

**Comment by carltongibson** [comment]
Yes, I'm seeing this this morning, on macOS too. No idea why yet. (It works in Docker...)

**Comment by carltongibson** [comment]
I'm wondering if this is https://bugs.python.org/issue33725

**Comment by carltongibson** [comment]
This should have been resolved in #247.

**Comment by ckcollab** [comment]
thanks @carltongibson :)
